src/main/java/net/snowflake/ingest/streaming/internal/SnowflakeStreamingIngestClientInternal.java

```diff
@@ -405,6 +405,25 @@ public SnowflakeStreamingIngestChannelInternal<?> openChannel(OpenChannelRequest
                     : ParquetProperties.WriterVersion.PARQUET_1_0)
             .build();
 
+    FullyQualifiedTableName fullyQualifiedTableName =
+        new FullyQualifiedTableName(
+            response.getDBName(), response.getSchemaName(), response.getTableName());
+    Long encryptionKeyId = response.getEncryptionKeyId();
+    String encryptionKey = response.getEncryptionKey();
+    if (encryptionKeyId != null && encryptionKey != null && !encryptionKey.isEmpty()) {
```

Comment:
Can `encryptionKeyId` be null?

Reply:
It should not be, this is just to be extra safe.